WcaOnRails/app/models/competition_event.rb

```diff
@@ -3,4 +3,19 @@ class CompetitionEvent < ApplicationRecord
   belongs_to :competition
   belongs_to :event
   has_many :registration_competition_events, dependent: :destroy
+  has_many :rounds, -> { order(:nth_round) }
+  accepts_nested_attributes_for :rounds, allow_destroy: true
+
+  validate do
+    remaining_rounds = rounds.reject(&:marked_for_destruction?)
```

Comment:
> maybe all ORMs?

Not all ORMs have validation baked right in, so this feels like an ActiveRecord thing going on.